File: pytwist/runAuditJob.py
#!/opt/opsware/agent/bin/python
# run script on server.py

import sys
sys.path.append("/opt/opsware/agent/pylibs")
from pytwist import *
from pytwist.com.opsware.search import Filter


# Create a TwistServer object.
ts = twistserver.TwistServer()
# Authenticate by passing an SA username and password
try:
 	ts.authenticate('admin','opsware_admin')
except:
	print "Authentication failed."
	sys.exit(2)


# Construct a search filter.
filter = Filter()
filter.expression = 'ServerVO.name *=* "%s"' % 'rhel55'
filter.objectType = 'device'

# Get a reference to ServerService.
serverservice = ts.server.ServerService
# Perform the search, returning server references.
serverRef = serverservice.findServerRefs(filter)
print "-------server %s" % serverRef

# Contruct a search filter.
filter = Filter()
filter.expression = 'AuditTaskVO.name *=* "%s"' % 'policy1'
filter.objectType = 'audit_task'

# Get a reference to AuditTaskService.
auditservice = ts.compliance.sco.AuditTaskService
# Perform the search, returning server references.
auditRef = auditservice.findAuditTaskRefs(filter)[0]
print "----------auditRef %s" % auditRef

# Run audit job
jobRef = auditservice.startAudit(auditRef,serverRef, None, None, None)
print "---------jobRef %s" % jobRef

File: java/source/runAuditJob.java
package source;

import com.opsware.client.OpswareClient;
import com.opsware.compliance.sco.*;
import com.opsware.job.JobRef;
import com.opsware.search.Filter;

public class runAuditJob {

	// @TODO read info from properties file
	// Connection details to the Opsware SAS core server.
	private static String host = "owc";
	private static int port = 443;
	private static String USERNAME = "admin";
	private static String PASSWD = "opsware_admin";

	// Saved script name
	private static String auditName = "policy1";
	private static AuditTaskRef auditRef;

	// Handles to Service.
	private static AuditTaskService auditTaskSvc;

	protected static void login() throws Exception {

		// Use the OpswareClient to create an https connection to Opsware SAS
		// and authenticate the user.
		OpswareClient.connect("https", host, (short) port, USERNAME, PASSWD,
				true);

		auditTaskSvc = (AuditTaskService) OpswareClient
				.getService(AuditTaskService.class);
	}


	protected static void getAuditTaskRef() throws Exception {
		String filterExpr = "AuditTaskVO.name *=* " + auditName;
		Filter filter = new Filter("audit_task", filterExpr);
		// Get references to the servers that match the search filter.
		auditRef = auditTaskSvc.findAuditTaskRefs(filter)[0];
		System.out.println("-------audit policy " + auditRef.toString());
	}

	protected static void runAuditTask() throws Exception {
		JobRef jobRef = auditTaskSvc.startAudit(auditRef, null, null, null);
		System.out.println("-------audit job " + jobRef.toString());
	}

	public static void main(String[] args) {
		try {
			login();
			getAuditTaskRef();
			runAuditTask();

		} catch (Exception e) {
			e.printStackTrace();
			System.out.println(e.getMessage());
			System.exit(1);
		}
	}

}
